go/oasis-test-runner/scenario/e2e/runtime/runtime_upgrade.go

```diff
@@ -0,0 +1,276 @@
+package runtime
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/oasisprotocol/oasis-core/go/common"
+	"github.com/oasisprotocol/oasis-core/go/common/sgx"
+	keymanager "github.com/oasisprotocol/oasis-core/go/keymanager/api"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis/cli"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario"
+	registry "github.com/oasisprotocol/oasis-core/go/registry/api"
+)
+
+// RuntimeUpgrade is the runtime upgrade scenario.
+var RuntimeUpgrade scenario.Scenario = newRuntimeUpgradeImpl()
+
+type runtimeUpgradeImpl struct {
+	runtimeImpl
+
+	nonce uint64
+}
+
+func newRuntimeUpgradeImpl() scenario.Scenario {
+	return &runtimeUpgradeImpl{
+		runtimeImpl: *newRuntimeImpl(
+			"runtime-upgrade",
+			"simple-keyvalue-enc-client",
+			nil,
+		),
+	}
+}
+
+func (sc *runtimeUpgradeImpl) Fixture() (*oasis.NetworkFixture, error) {
+	f, err := sc.runtimeImpl.Fixture()
+	if err != nil {
+		return nil, err
+	}
+
+	// Get number of compute runtimes.
+	var computeIndex int
+	for i := range f.Runtimes {
+		if f.Runtimes[i].Kind == registry.KindCompute {
+			computeIndex = i
+			break
+		}
+	}
+
+	// Load the upgraded runtime binary.
+	newRuntimeBinary, err := sc.resolveRuntimeBinary("simple-keyvalue-upgrade")
+	if err != nil {
+		return nil, fmt.Errorf("error resolving binary: %w", err)
+	}
+
+	// Setup the upgraded runtime (first is keymanager, others should be generic compute).
+	runtimeFix := f.Runtimes[computeIndex]
+	if runtimeFix.Kind != registry.KindCompute {
+		return nil, fmt.Errorf("expected second runtime in fixture to be compute runtime, got: %s", runtimeFix.Kind)
+	}
+	runtimeFix.Binaries = append([]string{newRuntimeBinary}, runtimeFix.Binaries...)
+
+	// The upgraded runtime will be registered later.
+	runtimeFix.ExcludeFromGenesis = true
+	newComputeIndex := len(f.Runtimes)
+	f.Runtimes = append(f.Runtimes, runtimeFix)
+
+	// Add the upgraded compute rutimes, will be started later.
+	workers := len(f.ComputeWorkers)
+	for i := range f.ComputeWorkers {
```

Comment:
What about storage workers, those have runtime-specific provisions as well? I guess this doesn't matter as the runtime identifiers are duplicated, but would probably be cleaner to support a `Runtimes` option for storage workers as well?

Reply:
This was resolved, but it seems like the storage workers are still not updated here in the fixture?